tests/integration/tls/test_tls.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+from tests.integration.helpers import (
+    APP_NAME,
+    SERIES,
+    UNIT_IDS,
+    get_application_unit_ips_names,
+    get_application_unit_names,
+    get_leader_unit_ip,
+)
+from tests.integration.tls.helpers import (
+    check_cluster_formation_successful,
+    check_security_index_initialised,
+    check_unit_tls_configured,
+)
+
+logger = logging.getLogger(__name__)
+
+TLS_CERTIFICATES_APP_NAME = "tls-certificates-operator"
+
+
+@pytest.mark.tls_tests
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+async def test_build_and_deploy_active(ops_test: OpsTest) -> None:
+    """Build and deploy one unit of MongoDB."""
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.set_config(
+        {
+            "logging-config": "<root>=INFO;unit=DEBUG",
+            "update-status-hook-interval": "1m",
+            "cloudinit-userdata": """postruncmd:
+                - [ 'sysctl', '-w', 'vm.max_map_count=262144' ]
+                - [ 'sysctl', '-w', 'fs.file-max=1048576' ]
+                - [ 'sysctl', '-w', 'vm.swappiness=0' ]
+                - [ 'sysctl', '-w', 'net.ipv4.tcp_retries2=5' ]
+        """,
+        }
+    )
+
+    await ops_test.model.deploy(
+        my_charm,
+        num_units=len(UNIT_IDS),
+        series=SERIES,
+    )
+    await ops_test.model.wait_for_idle()
+    assert len(ops_test.model.applications[APP_NAME].units) == len(UNIT_IDS)
+
+    # Deploy TLS Certificates operator.
+    config = {"generate-self-signed-certificates": "true", "ca-common-name": "CN_CA"}
+    await ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="edge", config=config)
+    await ops_test.model.wait_for_idle(
+        apps=[TLS_CERTIFICATES_APP_NAME], status="active", timeout=1000
+    )
+
+    # Relate it to OpenSearch to set up TLS.
+    await ops_test.model.relate(APP_NAME, TLS_CERTIFICATES_APP_NAME)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=5000)
```

Comment:
Just noting, this `wait_for_idle` will almost always pass due to bugs with how Juju exposes status with `pytest-operator`.

If you have another way of checking a 'healthy' tls relation, even if it's exec-ing a `juju show-unit` and pulling out some relation-data

Reply:
If that's what you're doing below, resolve this.
Just concerned that `active` status isn't a reliable start-point/end-point before taking other actions that depend on the completion of something in that relation, as it can evaluate `active` even during the relation events.

Reply:
Thanks Marc, that's good to know. Juju should be on a BlockedState unless TLS is fully configured. Luckily, if it's buggy - I am still testing the correctness of the TLS conf in the 3 following tests, `test_security_index_initialised` - `test_tls_configured` - `test_cluster_formation_after_tls`